packages/superset-ui-encodable/src/encoders/Encoder.ts

```diff
@@ -92,6 +95,55 @@ export default class Encoder<Config extends EncodingConfig> {
     return Array.from(new Set(fields));
   }
 
+  private createLegendItemsFactory(field: string) {
+    const channelEncoders = flatMap(
+      this.getChannelEncoders().filter(e => isNotArray(e) && isValueDef(e.definition)),
+    ).concat(this.legends[field]);
+
+    return (domain: ChannelInput[]) =>
+      domain.map((input: ChannelInput) => ({
+        input,
+        output: channelEncoders.reduce(
+          (prev: Partial<{ [k in keyof Config]: Config[k]['1'] }>, curr) => {
+            const map = prev;
+            map[curr.name as keyof Config] = curr.encodeValue(`${input}`) as Value;
+
+            return map;
+          },
+          {},
+        ),
+      }));
+  }
+
+  getLegendInformation(data: Dataset = []) {
+    return (
+      Object.keys(this.legends)
+        // for each field that was encoded
+        .map((field: string) => {
+          // get all the channels that use this field
+          const channelEncoders = this.legends[field];
+          const firstEncoder = channelEncoders[0];
+          const definition = firstEncoder.definition as TypedFieldDef;
+          const createLegendItems = this.createLegendItemsFactory(field);
+
+          if (definition.type === 'nominal') {
+            return {
+              createLegendItems,
+              field,
+              items: createLegendItems(firstEncoder.getDomain(data)),
+              type: 'nominal' as const,
```

Comment:
could set as `definition.type` like below, or could refactor to ternary just for `items` field

Reply:
can you also add a comment for why we can't create items for non-nominal types?

Reply:
what's the advantage of creating the `items` if the consumer can just invoke `createLegendItems` anyway and it only works if they pass `data`? seems simpler to just have a consistent behavior for different types?

Reply:
We can create items for non-nominal types but it will take time to implement. basically getting the right `ticks` from `domain` for linear, quantize, quantile, log, etc. scale. I kind of punt it for now.